src/Components/Components/test/Routing/RouterTest.cs

```diff
@@ -42,72 +42,39 @@ public async Task CanRunOnNavigateAsync()
         {
             // Arrange
             var called = false;
-            async Task OnNavigateAsync(NavigationContext args)
+            Action<NavigationContext> OnNavigateAsync = async (NavigationContext args) =>
             {
                 await Task.CompletedTask;
                 called = true;
-            }
-            _router.OnNavigateAsync = OnNavigateAsync;
-
-            // Act
-            await _renderer.Dispatcher.InvokeAsync(() => _router.RunOnNavigateWithRefreshAsync("http://example.com/jan", false));
-
-            // Assert
-            Assert.True(called);
-        }
-
-        [Fact]
-        public async Task CanHandleSingleFailedOnNavigateAsync()
```

Comment:
Why are we removing coverage for this? Shouldn't we still test what happens when an exception is thrown from the handler?

Reply:
We have an E2E test that validates this scenario [here](https://github.com/dotnet/aspnetcore/blob/master/src/Components/test/testassets/BasicTestApp/RouterTest/TestRouterWithOnNavigate.razor#L53) so I removed this.

Reply:
That's fair, for future reference I would suggest you consider keeping the unit test and the E2E test. They complement each other.

The unit test validates the component and helps catch regressions faster while the E2E test validates the integration. Figuring out a regression with a failing E2E test is much harder than with a unit test.

Reply:
Turns out that doing this in the E2E tests is the only option I have. There isn't a super nice way to create an EventCallback with the Router component instantiated with the tests (hence why we use `new EventCallback(null, action)`. This exercises the execution of the async action but doesn't exercise the renderer.

Reply:
Hmm, I do know how to do this, but it requires a bit of changes. It involves:
* Creating a parent component.
* Attaching the instance to the renderer.
* Setting properties on the parent component instance.
* Triggering a render from the parent component.

There are a lot of samples in the Renderer tests, but it is a bit more involved.

Reply:
I'll let you use your best judgement here.